src/components/UserSelector.tsx

```diff
@@ -1,16 +1,47 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { User } from '../types/User';
+import { getUsers } from './api/users';
+import classNames from 'classnames';
+import { Post } from '../types/Post';
+
+type Props = {
+  onSelectedUser: (val: User | null) => void;
+  selectedUser: User | null;
+  onSelectedPost: (val: Post | null) => void;
+};
+
+export const UserSelector: React.FC<Props> = ({
+  selectedUser,
+  onSelectedUser,
+  onSelectedPost,
+}) => {
+  const [users, setUsers] = useState<User[]>([]);
```

Comment:
You are initializing state with empty array. It's not a mistake, but it's better to initialize state with `null` to know exactly when data hasn't been loaded yet. It helps to avoid unnecessary renders and makes code more predictable.